reana_commons/api_client.py

```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of REANA.
+# Copyright (C) 2018 CERN.
+#
+# REANA is free software; you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation; either version 2 of the License, or (at your option) any later
+# version.
+#
+# REANA is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# REANA; if not, write to the Free Software Foundation, Inc., 59 Temple Place,
+# Suite 330, Boston, MA 02111-1307, USA.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization or
+# submit itself to any jurisdiction.
+"""REANA REST API base client."""
+
+import json
+import os
+
+import pkg_resources
+from bravado.client import SwaggerClient
+
+
+class BaseAPIClient(object):
+    """REANA API client code."""
+
+    def __init__(self, component, server_data, http_client=None):
```

Comment:
Why we need the `component` parameter? I can not see where it is used

Reply:
I am thinking 🤔 . Since we centralise the OpenAPI specs here, we could also centralise the configuration for OpenAPI clients here so instead of having this configuration in each component, [i.e. in `reana-server`](https://github.com/reanahub/reana-server/pull/93/files#diff-20447ed4c4c4479b80b32b3573144522R26):
```python
COMPONENTS_DATA = {
    'reana-workflow-controller': (
       'http://{address}:{port}'.format(
           address=os.getenv('WORKFLOW_CONTROLLER_SERVICE_HOST', '0.0.0.0'),
           port=os.getenv('WORKFLOW_CONTROLLER_SERVICE_PORT_HTTP', '5000')),
       'reana_workflow_controller.json'),
}
```
We could have a mapping here in commons which would look like:
```python
OPENAPI_COMPONENTS = {
    'reana-workflow-controller': (
        'http://{address}:{port}'.format(
            address=os.getenv('WORKFLOW_CONTROLLER_SERVICE_HOST', '0.0.0.0'),
            port=os.getenv('WORKFLOW_CONTROLLER_SERVICE_PORT_HTTP', '5000')),
        'reana_workflow_controller.json'),
    'reana-server': (
        os.getenv('REANA_SERVER_URL', None),
        'reana_server.json'),
    'another-component': (url, spec)
    ...
}
```
So the APIClient gets the name of the component you want to build the API client for, and through the mapping it gets the specific `(url, spec)`. Moreover, the knowledge of in which path is the spec and how the configuration is retrieved from the environment remains in `reana-commons` (I already saw that we retrieve the config from environment different from `reana-client` and for `reana-server`). What do you think?